lib/travis/build/script/go.rb

```diff
@@ -149,7 +150,7 @@ def comparable_go_version
           end
 
           def go_get_cmd
-            (go_version == 'go1' || comparable_go_version < Gem::Version.new('1.2')) ? 'go get' : 'go get -t'
+            (go_version == 'go1' || (go_version != 'tip' && comparable_go_version < Gem::Version.new('1.2'))) ? 'go get' : 'go get -t'
```

Comment:
After a bit of pondering on this, I think I've come to the conclusion that using helpers to group versions is both hard to understand ("Why are they asking about specific versions here?") and is just going to make things worse over time, as more versions get special treatment.

I think the best way to handle this is to provide, like I did for vendoring, a special helper that answers the question at hand. In this case, a `can_get_test_dependencies?` helper that returns `true` if there's a `-t` flag to use with `go get` to retrieve dependencies for tests. Then this becomes:

```ruby
(can_get_test_dependencies?) ? 'go get -t' : 'go get'
```

Which I think is much more readable and clearer in what it's actually asking. What do you think?